substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -0,0 +1,1591 @@
+/// h2   -> [ Ok(xth0),    Err,         Err       ]
+/// h3   -> [ Ok(xth0),    Ok(xth1),    Err       ]
+///
+/// out:
+/// [ Ok(xth0), Ok(xth1), Err ]
+/// ```
+fn reduce_multiview_result<H, E>(input: HashMap<H, Vec<Result<H, E>>>) -> Vec<Result<H, E>> {
+	let mut values = input.values();
+	let Some(first) = values.next() else {
+		return Default::default();
+	};
+	let length = first.len();
+	debug_assert!(values.all(|x| length == x.len()));
+
+	input
+		.into_values()
+		.reduce(|mut agg_results, results| {
+			agg_results.iter_mut().zip(results.into_iter()).for_each(|(agg_r, r)| {
+				if agg_r.is_err() {
+					*agg_r = r;
+				}
+			});
+			agg_results
+		})
+		.unwrap_or_default()
+}
+
+impl<ChainApi, Block> TransactionPool for ForkAwareTxPool<ChainApi, Block>
+where
+	Block: BlockT,
+	ChainApi: 'static + graph::ChainApi<Block = Block>,
+	<Block as BlockT>::Hash: Unpin,
+{
+	type Block = ChainApi::Block;
+	type Hash = ExtrinsicHash<ChainApi>;
+	type InPoolTransaction = Transaction<ExtrinsicHash<ChainApi>, ExtrinsicFor<ChainApi>>;
+	type Error = ChainApi::Error;
+
+	/// Submits multiple transactions and returns a future resolving to the submission results.
+	///
+	/// Actual transactions submission process is delegated to the `ViewStore` internal instance.
+	///
+	/// The internal limits of the pool are checked. The results of submissions to individual views
+	/// are reduced to single result. Refer to `reduce_multiview_result` for more details.
+	fn submit_at(
+		&self,
+		_: <Self::Block as BlockT>::Hash,
+		source: TransactionSource,
+		xts: Vec<TransactionFor<Self>>,
+	) -> PoolFuture<Vec<Result<TxHash<Self>, Self::Error>>, Self::Error> {
+		let view_store = self.view_store.clone();
+		log::debug!(target: LOG_TARGET, "fatp::submit_at count:{} views:{}", xts.len(), self.active_views_count());
+		log_xt_trace!(target: LOG_TARGET, xts.iter().map(|xt| self.tx_hash(xt)), "[{:?}] fatp::submit_at");
+		let xts = xts.into_iter().map(Arc::from).collect::<Vec<_>>();
+		let mempool_result = self.mempool.extend_unwatched(source, xts.clone());
+
+		if view_store.is_empty() {
+			return future::ready(Ok(mempool_result)).boxed()
+		}
+
+		let (hashes, to_be_submitted): (Vec<TxHash<Self>>, Vec<ExtrinsicFor<ChainApi>>) =
+			mempool_result
+				.iter()
+				.zip(xts)
+				.filter_map(|(result, xt)| result.as_ref().ok().map(|xt_hash| (xt_hash, xt)))
+				.unzip();
+
+		self.metrics
+			.report(|metrics| metrics.submitted_transactions.inc_by(to_be_submitted.len() as _));
+
+		let mempool = self.mempool.clone();
+		async move {
+			let results_map = view_store.submit(source, to_be_submitted.into_iter(), hashes).await;
+			let mut submission_results = reduce_multiview_result(results_map).into_iter();
+
+			//todo [#5494]:
+			//ImmediatelyDropped errors from view submission shall be ignored. If transaction got into the mempool,
+			//(and not into the view) it means that it was successfully submitted. It will be sent
```

Comment:
This will need one more iteration for proper handling of priorities (https://github.com/paritytech/polkadot-sdk/issues/5809). Leaving it as is.

Reply:
> Also generally, will it ever go to the mempool when no view accepted it? I think no?

Currenty it is placed into mempool first - so yes.
Also `ImmediatelyDropped` does not mean that tx is invalid - it was just dropped because of limits.